automation/jleechanorg_pr_automation/openai_automation/codex_github_mentions.py

```
@@ -124,15 +152,71 @@ async def connect_to_existing_browser(self) -> bool:
+            if self.page:
+                print("📄 Reusing existing tab for automation")
+            else:
+                self.page = await self.context.new_page()
+                print("📄 Created new page for automation")
             return True
         except Exception as e:
             print(f"❌ Failed to connect via CDP: {e}")
             logger.warning(f"CDP connection failed: {e}")
             return False
 
+    async def _select_existing_page(self) -> Optional[Page]:
```

Ruff (0.14.10)

Add `from __future__ import annotations` to simplify `typing.Optional`

Add `from __future__ import annotations`

(FA100)